File: Week_01/G20190343020055/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    for (let i = 0; i < nums.length; i++) {

        let element = nums[i]
        let b = target - element
        for (let j = i+1; j < nums.length; j++) {

            if (b == nums[j]) return [i, j]
        }
    }
    return []
};

File: Week_02/G20190343020055/two-sum.go

func twoSum(nums []int, target int) []int {

	m := make(map[int]int)

	for index := 0; index < len(nums); index++ {

		b := target - nums[index]

		if j, ok := m[b]; ok {

			return []int{j, index}
		} else {
			m[nums[index]] = index
		}
	}

	return nil
}
